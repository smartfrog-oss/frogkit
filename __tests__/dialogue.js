import { mount, shallow } from 'vue-test-utils'
import Dialogue from '@/components/dialogue'

const propsData = {
  visible: true,
  dark: true,
  requiredAction: false
}

const slotData = {
  emitEvent: 'testEvent',
  emitValue: 'xxx'
}

describe('Dialogue component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(Dialogue, { propsData })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit change event when close method is called', () => {
    const cmp = shallow(Dialogue, { propsData })
    const stub = jest.fn()
    cmp.vm.$on('change', stub)
    cmp.vm.close()
    expect(stub).toBeCalled()
  })
})
