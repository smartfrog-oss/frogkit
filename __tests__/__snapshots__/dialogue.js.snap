// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogue component should render component and match snapshot 1`] = `
<flex justify="center" align="center" class="fk-dialogue fk-dialogue--dark v-enter fk-dialogue__animation-in">
  <section class="fk-dialogue__frame"><a class="fk-dialogue__close">
      <icon icon="close-light" color="orange"></icon>
    </a> </section>
</flex>
`;
